packages/template-chakra-storefront/src/components/with-translations/index.jsx

```diff
@@ -0,0 +1,122 @@
+// hoc/withTranslations.js
+import {useIntl} from 'react-intl'
+
+// Re-export defineMessages for a cohesive API. This allows components to import
+// everything they need for translations from a single file, simplifying imports
+// and making mocking for tests.
+// WARNING: you MUST keep the name "defineMessages" for react-intl static analysis to work.
+export {defineMessages} from 'react-intl'
```

Comment:
This is important thing to highlight.

Here, I'm **re-exporting** the `defineMessages` from react-intl, the reason for this is I want to simplify the import statements from react components.

BEFORE
```
// you always need to import 2 things
import {defineMessages} from 'react-intl'
import {withTranslations} from './components/with-translations'
```

AFTER
```
import {withTranslations, defineMessages} from './components/with-translations'
```

Why is this needed?

1. cleaner shorter import statements
2. less mocking you need for your component unit tests

Reply:
Do we need to export other APIs?

No, `defineMessages` + `withTranslations` should cover all use cases.